### DISCUSSION TITLE
Running Chrome on Fargate

### DISCUSSION THREAD

**Discussion by neil-snapstream** [artilleryio/artillery#2435] [Artillery]
By setting the `playwright > launchOptions > channel` option to "chrome", I am able to run artillery using chrome locally.  I have to run my playwright tests with chrome instead of chromium because chromium doesn't support HLS playback. However, when running on Fargate, I get the error:

```
worker error, id: 1 browserType.launch: Chromium distribution 'chrome' is not found at /opt/google/chrome/chrome
Run "npx playwright install chrome"
```

Is there a way to install chrome on the ecs containers that `artillery run-fargate` creates before executing the playwright script?

**Comment by hassy** [comment]
hi @neil-snapstream 👋 the only way to do it right now would be to build a custom Docker image using [Artillery's Dockerfile](https://github.com/artilleryio/artillery/blob/main/packages/artillery/lib/platform/aws-ecs/worker/Dockerfile) as the base, publish it somewhere, and then override the default image used on Fargate with `WORKER_IMAGE_URL` environment variable, e.g.:

```sh
WORKER_IMAGE_URL=myregistry/custom-image artillery run-fargate myscript.yml
```

we'd like to introduce the ability to do this out of the box at some point, but there's no ETA for that yet.

**Reply by neil-snapstream** [reply]
Thanks @hassy

**Reply by sridharveeravalli** [reply]
Hello @hassy - I followed your advice and built a Docker image based on the code here - 
https://github.com/artilleryio/artillery/tree/main/packages/artillery/lib/platform/aws-ecs/worker

I am able to run examples for browser load testing in headless mode. 
But I get error if I set headless mode = false. 

Is there a way to run artillery script with playwright test on a browser inside a Docker container? 


Error:
![Uploading Screenshot 2024-05-10 at 4.53.03 PM.png…]()

**Reply by hassy** [reply]
@sridharveeravalli will need more information to try to help. What changes did you make to the Docker image? And are you seeing any errors, stack traces etc in the logs on CloudWatch?

**Reply by sridharveeravalli** [reply]
1. Update browser-load-test to include a proxy and set headless = false
2. Unable to attach Docker file here. Made updates to get over the certificate errors for npm packages.
I like to know if playwright tests can be run on fargate Docker container using Artillery. If yes, then how to do it.

Error message:
⠦ worker error, id: 1 browserType.launch: Target page, context or browser has been closed
Browser logs:

╔════════════════════════════════════════════════════════════════════════════════════════════════╗
║ Looks like you launched a headed browser without having a XServer running.                     ║
║ Set either 'headless: true' or use 'xvfb-run <your-playwright-app>' before running Playwright. ║
║                                                                                                ║
║ <3 Playwright Team                                                                             ║
╚════════════════════════════════════════════════════════════════════════════════════════════════╝
Call log:
  - <launching> /ms-playwright/chromium-1112/chrome-linux/chrome --disable-field-trial-config --disable-background-networking --enable-features=NetworkService,NetworkServiceInProcess --disable-background-timer-throttling --disable-backgrounding-occluded-windows --disable-back-forward-cache --disable-breakpad --disable-client-side-phishing-detection --disable-component-extensions-with-background-pages --disable-component-update --no-default-browser-check --disable-default-apps --disable-dev-shm-usage --disable-extensions --disable-features=ImprovedCookieControls,LazyFrameLoading,GlobalMediaControls,DestroyProfileOnBrowserClose,MediaRouter,DialMediaRouteProvider,AcceptCHFrame,AutoExpandDetailsElement,CertificateTransparencyComponentUpdater,AvoidUnnecessaryBeforeUnloadCheckSync,Translate,HttpsUpgrades,PaintHolding --allow-pre-commit-input --disable-hang-monitor --disable-ipc-flooding-protection --disable-popup-blocking --disable-prompt-on-repost --disable-renderer-backgrounding --force-color-profile=srgb --metrics-recording-only --no-first-run --enable-automation --password-store=basic --use-mock-keychain --no-service-autorun --export-tagged-pdf --disable-search-engine-choice-screen --no-sandbox --enable-precise-memory-info --disable-dev-shm-usage --user-data-dir=/tmp/playwright_chromiumdev_profile-XXXXXXoCJHzc --remote-debugging-pipe --no-startup-window
  - <launched> pid=18
  - [pid=18][err] [18:32:0509/165506.971826:ERROR:bus.cc(407)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
  - [pid=18][err] [18:18:0509/165506.975227:ERROR:ozone_platform_x11.cc(243)] Missing X server or $DISPLAY
  - [pid=18][err] [18:18:0509/165506.975279:ERROR:env.cc(258)] The platform failed to initialize.  Exiting.

**Reply by sridharveeravalli** [reply]
@hassy - any updates?

**Reply by hassy** [reply]
You can't run a headed browser on Fargate. Also unless you're trying to use a browser that's not Chromium you don't need to build a custom Docker image.
